stage0/run.go

```diff
@@ -165,20 +159,78 @@ func Prepare(cfg PrepareConfig, dir string, uuid *types.UUID) error {
 
 	cdoc, err := json.Marshal(cm)
 	if err != nil {
-		return fmt.Errorf("error marshalling pod manifest: %v", err)
+		return nil, fmt.Errorf("error marshalling pod manifest: %v", err)
 	}
+	return cdoc, nil
+}
 
-	log.Printf("Writing pod manifest")
-	fn := common.PodManifestPath(dir)
-	if err := ioutil.WriteFile(fn, cdoc, 0700); err != nil {
-		return fmt.Errorf("error writing pod manifest: %v", err)
+// validatePodManifest reads the use specified pod manifest, prepare the app images
+// and do validation for the pod manifest. If the pod manifest looks good, returns
+// the manifest as []byte.
+// TODO(yifan): More validation in the future.
+func validatePodManifest(cfg PrepareConfig, dir string) ([]byte, error) {
+	cdoc, err := ioutil.ReadFile(cfg.PodManifest)
+	if err != nil {
+		return nil, fmt.Errorf("error reading pod manifest: %v", err)
```

Comment:
this is actually a case where you probably just want to return the raw error, "error reading pod manifest" is misleading and superfluous